tests-legacy/lib/rules/assertion-before-screenshot.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const rule = require('../../../lib/rules/assertion-before-screenshot')
+const RuleTester = require('eslint').RuleTester
+
+const ruleTester = new RuleTester()
+
+const errors = [{ messageId: 'unexpected' }]
+const parserOptions = { ecmaVersion: 6 }
+
+ruleTester.run('assertion-before-screenshot', rule, {
```

Comment:
@MikeMcC399 I'm not sure how the new tester works, but if possible, it seems like there'd be utility to breaking out these rules so that they can be in a shared file used for both legacy and recent tests. Just thinking how manual it will be to update both sets of tests whenever an update it needed. Not really sure it's worth it though for just 2 files.

Reply:
@jennifer-shehane 

This is similar to the Cypress migration from `9.x` to `10.x` where it was necessary in Cypress GitHub Actions to have two sets of tests in parallel for a limited time, one set for Cypress `9.x` that was only updated as absolutely necessary and a new set of tests for Cypress `10.x` which were maintained and updated. At some stage, a new major version was declared and support for Cypress `9.x` was dropped.

It will be difficult to continue to support ESLint `7`, `8` and `9` all in one version of this plugin and at some stage support for earlier versions will need to be dropped. I will put this is a separate issue for discussion.

Thanks for your comments!